src/components/split-button/split-button-rtl.test.tsx

```diff
@@ -0,0 +1,912 @@
+    const user = userEvent.setup();
+    const onKeyDownMock = jest.fn();
+    render(
+      <SplitButton text="Main" onKeyDown={onKeyDownMock}>
+        {singleButton()}
+      </SplitButton>
+    );
+
+    const main = await screen.findByRole("button", { name: "Main" });
+    await user.tab();
+
+    expect(main).toHaveFocus();
+    await user.keyboard("{Enter}");
+    expect(onKeyDownMock).toHaveBeenCalled();
+    expect(
+      screen.queryByRole("button", { name: "Single Button" })
+    ).not.toBeInTheDocument();
+  });
+
+  it("should not render the child buttons when a 'Space' keydown event detected and main button is focused", async () => {
+    const user = userEvent.setup();
+    const onKeyDownMock = jest.fn();
+    render(
+      <SplitButton text="Main" onKeyDown={onKeyDownMock}>
+        {singleButton()}
+      </SplitButton>
+    );
+
+    const main = await screen.findByRole("button", { name: "Main" });
+    await user.tab();
+
+    expect(main).toHaveFocus();
+    await user.keyboard("{space}");
+    expect(onKeyDownMock).toHaveBeenCalled();
+    expect(
+      screen.queryByRole("button", { name: "Single Button" })
+    ).not.toBeInTheDocument();
+  });
+
+  it("should not render the child buttons when main button is clicked", async () => {
+    const user = userEvent.setup();
+    const onClickMock = jest.fn();
+    render(
+      <SplitButton text="Main" onClick={onClickMock}>
+        {singleButton()}
+      </SplitButton>
+    );
+
+    const main = await screen.findByRole("button", { name: "Main" });
+    await user.click(main);
+
+    expect(onClickMock).toHaveBeenCalled();
+    expect(
+      screen.queryByRole("button", { name: "Single Button" })
+    ).not.toBeInTheDocument();
+  });
+
+  it("should not call the `onClick` handle when main button is clicked whilst disabled", async () => {
+    const user = userEvent.setup();
+    const onClickMock = jest.fn();
+    render(
+      <SplitButton text="Main" onClick={onClickMock} disabled>
+        {singleButton()}
+      </SplitButton>
+    );
+
+    await user.click(screen.getByRole("button", { name: "Main" }));
+
+    expect(onClickMock).not.toHaveBeenCalled();
+  });
+
+  it("should hide the additional buttons when a click event detected outside the component", async () => {
+    const user = userEvent.setup();
+    render(<SplitButton text="Main">{singleButton()}</SplitButton>);
+
+    const toggle = screen.getByRole("button", { name: "Show more" });
+    await user.click(toggle);
+    const childButton = screen.getByRole("button", { name: "Single Button" });
+
+    expect(childButton).toBeInTheDocument();
+
+    await waitFor(async () => {
+      await user.click(document.body);
+
+      expect(screen.queryByRole("list")).not.toBeInTheDocument();
+    });
+  });
+
+  it("should hide the additional buttons when a mouseout event detected on the toggle button", async () => {
```

Comment:
I'm not sure what you mean here, it's not a mouseover event. It could be argued it's a mouseleave but I don't think the semantics are so different here that it's not evidently clear what the test is asserting etc